components/movementsensor/imuwit/imu.go

```diff
@@ -202,12 +203,20 @@ func (imu *wit) Position(ctx context.Context, extra map[string]interface{}) (*ge
 
 func (imu *wit) Accuracy(ctx context.Context, extra map[string]interface{}) (*movementsensor.Accuracy, error,
 ) {
-	// TODO: RSDK-6389 return the compass heading from the datasheet of the witIMU if the pitch angle is less than 45 degrees
+	// return the compass heading from the datasheet (0.5) of the witIMU if the pitch angle is less than 45 degrees
 	// and the roll angle is near zero
 	// mag projects at angles over this threshold cannot be determined because of the larger contribution of other
 	// orientations to the true compass heading
 	// return NaN for compass accuracy otherwise.
-	return movementsensor.UnimplementedAccuracies()
+
+	roll := imu.orientation.Roll
+	pitch := imu.orientation.Pitch
+
+	if math.Abs(roll) <= 1 && math.Abs(pitch) <= maxTiltInRad {
+		return &movementsensor.Accuracy{CompassDegreeError: float32(compassAccuracy)}, nil
+	}
+
+	return &movementsensor.Accuracy{CompassDegreeError: float32(math.NaN())}, nil
```

Comment:
We might need a lock now since we're accessing the sturct , check if polling this function from different client scripts at the same time the RC card is healthy, or any races from Reconfiguration/Close.